File: solutions/3046. Split the Array/3046.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
public:
  bool isPossibleToSplit(const vector<int>& nums) {
    vector<int> count(101);

    for (const int num : nums) {
      ++count[num];
      if (count[num] > 2)
        return false;
    }

    return true;
  }
};

File: solutions/3046. Split the Array/3046.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def isPossibleToSplit(self, nums: List[int]) -> bool:
    count = [0] * 101

    for num in nums:
      count[num] += 1
      if count[num] > 2:
        return False

    return True  